collab-service/src/question_client/question_agent.ts

```diff
@@ -18,7 +26,7 @@ class QuestionAgent implements IQuestionAgent {
     );
   }
 
-  getQuestionByDifficulty(difficulty: QuestionDifficulty): Promise<Question | undefined> {
+  getQuestionByDifficulty(difficulty: QuestionDifficulty): Promise<GetQuestionResponse> {
```

Comment:
The previous typing `Question | undefined` was more logical. Is there a reason for this change?